### ISSUE TITLE
The Grep Test

### ISSUE THREAD

**Issue by rreusser** [dwmkerr/hacker-laws#106]
I'm not sure what the standard of inclusion is, but I'd love for there to be a counterpoint on DRY (ref: #19). I absorbed the concept of DRY and did a lot of damage very quickly. As I improve and slowly pick up the pieces though, The Grep Test became a great way for me to think about one of the most valuable and difficult things for me to internalize as a developer. See: [Too DRY - The Grep Test](http://jamie-wong.com/2013/07/12/grep-test/)

Would this be a welcome addition?

**Comment by siriak** [comment]
After reading references, I wonder if it can be an issue in statically typed languages? Examples were written in JS, Ruby and Python and I can see their point there, but it's very uncommon to encounter such code in statically typed languages. So I'm wondering if this principle is applicable in my case (C# developer). I'd really appreciate it if the author could give me some more further reading with examples in statically typed languages. Personally I welcome this addition as I think it's important not to be obsessed with principles and patterns and make decisions based on your situation.

**Comment by rreusser** [comment]
Good point! Hmm, maybe while it doesn’t technically fail the test, overloading the comma operator would be an example: https://mobile.twitter.com/matt77hias/status/1124285841453584388

While it doesn’t technically fail, it pushes the test to the breaking point because 1) I for one would stare at this blankly and not think to grep for `operator\s?,`, 2) I don’t think IDEs will show any meaningful completion or definition lookup, and 3) ‘.get(i, j)`, though admittedly not as cute, is performance-equivalent and way easier to grep and comprehend.

There may be more clear-cut examples since this doesn’t *technically* fail, but I consider it similar in spirit.

**Comment by rreusser** [comment]
Oh, that’s C++ and not C# but statically typed nonetheless. Also I wonder how far you take `#define` down the grep test failure road. 😬

**Comment by siriak** [comment]
Thanks for the example a lot :)
